src/cats/CatApp.js

```diff
@@ -0,0 +1,83 @@
+import { useEffect, useReducer } from "react";
+import useAbortableFetch from "./useAbortableFetch.js";
+import getCatsApiFetchParams from "./getCatsApiFetchParams.js";
+import CatSlideshow from "./CatSlideshow.js";
+import CatSlideshowControls from "./CatSlideshowControls.js";
+import BreedSelector from "./BreedSelector.js";
+import { MessageCard, LoadingCard } from "./Cards.js";
+import "./CatApp.css";
+
+const BREED_ID_KEY = "CatApp-breed-id";
+
+function catAppReducer(state, action) {
+  //console.log("%ccatapp: " + action.type, "color: green", action);
+  switch (action.type) {
+    case "select-breed": {
+      window.localStorage.setItem(BREED_ID_KEY, action.id);
+      return { ...state, selectedBreedID: action.id };
+    }
+    case "breeds-loaded":
+      return { breeds: action.breeds, selectedBreedID: action.selectedBreedID };
+    default:
+      throw new Error(`Unhandled action type: ${action.type}`);
+  }
+}
+
+function CatApp() {
+  const { status: loadStatus, runFetch } = useAbortableFetch("loading");
+  const [state, dispatch] = useReducer(catAppReducer, {
+    breeds: null,
+    selectedBreedID: null,
+  });
+
+  useEffect(() => {
+    const { url, options } = getCatsApiFetchParams("/breeds");
+
+    runFetch(url, options)
+      .then(async (response) => {
+        if (!response || !response.ok) {
+          throw new Error("API call failed");
+        }
+
+        const json = await response.json();
+        let breeds = [{ id: "all", name: "All Breeds" }].concat(
+          json.filter((b) => ({ id: b.id, name: b.name }))
+        );
+
+        dispatch({
+          type: "breeds-loaded",
+          breeds,
+          selectedBreedID: window.localStorage.getItem(BREED_ID_KEY) || "all",
+        });
+      })
+      .catch((error) => {
+        // suppress error
+      });
+  }, [runFetch]);
+
+  return (
+    <div className="CatApp">
+      <BreedSelector
+        dispatch={dispatch}
+        breeds={state.breeds}
+        selectedBreedID={state.selectedBreedID}
+        status={loadStatus}
+      />
+      {loadStatus === "loading" || state.selectedBreedID === null ? (
+        <>
+          <LoadingCard />
+          <CatSlideshowControls isDisabled={true} />
+        </>
+      ) : loadStatus === "error" ? (
+        <>
+          <MessageCard className="error">an error has occurred</MessageCard>
+          <CatSlideshowControls isDisabled={true} />
```

Comment:
I agree with you guys that CatSlideshow should own rendering the controls.

@chadwhiting, rather than pass loadStatus into the Cards, I'd prefer to keep them dumb and continue to have CatApp manage their display.